apis/v1alpha1/generator.yaml

```diff
@@ -22,11 +22,22 @@ operations:
   StopPipelineExecution:
     operation_type: Delete
     resource_name: PipelineExecution
+  AddTags:
+    operation_type: Update
+    resource_name: Model
```

Comment:
This generates code that treats `AddTags` as the Update API Call to update the model.

Reply:
@a-hilaly - Thanks for review.
Sagemaker model API do not have any modify /update method , as fields are immutable except tags . Also Sagemaker API resource only have "ADDTAG" and "DELETE TAG" . So to handle update on this resource will have to use one of these two. So chose "ADD TAGS" and handling "DELETE tags" using template hooks. 

Let me know if you think any other approach we can consider

Reply:
Agree with Amine, need to discuss this further as we need to throw appropriate error if any other fields except Tags are modified as Model does not have udpate API, maybe implement it completely via hooks

Reply:
@surajkota - Added check in update_pre hooks to validate field to update, if it is other than tag , reconciler will update "message" status.

Please let me know if you feel anything else is require